### ISSUE TITLE
Any plans to introduce compiler for in-memory data?

### ISSUE THREAD

**Issue by nikhilsarvaiye** [sqlkata/querybuilder#649]
Any plans to introduce compiler for in-memory data in lists, collections? which can be useful in case of caching like Redis, or for general code performance.

I would love to know your thoughts, if would like I can contribute, thoughts?

**Comment by ahmad-moussawi** [comment]
Nice idea!, but I don't think it's a matter of the compiler only. The job of a compiler is to convert C# statements to SQL string statements, so maybe a package with a broader scope would be needed in this case.

**Comment by nikhilsarvaiye** [comment]
Right, I would say you build Query and then Query to filter in-memory dictionaries. This way we can reuse the query object, I also have few ideas where and how joins can work in memory...

**Comment by Cricle** [comment]
I think DataTable.Select can use

**Comment by michaelkruglos** [comment]
Has anyone tried SQLite in-memory feature?
I've found [this](https://learn.microsoft.com/en-us/dotnet/standard/data/sqlite/in-memory-databases), but I haven't tried it yet.
It sure seems as a viable solution for unit tests, where you would use it for in-memory mock DB.

**Comment by ahmad-moussawi** [comment]
The unit tests already use this, I think you can follow the same approach

https://github.com/sqlkata/querybuilder/blob/master/QueryBuilder.Tests/SQLiteExecutionTest.cs
